uaclient/tests/test_daemon.py

```diff
@@ -0,0 +1,503 @@
+import mock
+import pytest
+
+from uaclient import exceptions, status, util
+from uaclient.clouds.aws import UAAutoAttachAWSInstance
+from uaclient.clouds.gcp import UAAutoAttachGCPInstance
+from uaclient.daemon import (
+    attempt_auto_attach,
+    poll_for_pro_license,
+    start,
+    stop,
+)
+
+M_PATH = "uaclient.daemon."
+
+
+@mock.patch(M_PATH + "util.subp")
+class TestStart:
+    def test_start_success(self, m_subp):
+        start()
+        assert [
+            mock.call(["systemctl", "start", "ubuntu-advantage.service"])
+        ] == m_subp.call_args_list
+
+    @mock.patch(M_PATH + "LOG.warning")
+    def test_start_warning(self, m_log_warning, m_subp):
+        err = util.ProcessExecutionError("cmd")
+        m_subp.side_effect = err
+        start()
+        assert [
+            mock.call(["systemctl", "start", "ubuntu-advantage.service"])
+        ] == m_subp.call_args_list
+        assert [mock.call(err)] == m_log_warning.call_args_list
+
+
+@mock.patch(M_PATH + "util.subp")
+class TestStop:
+    def test_stop_success(self, m_subp):
+        stop()
+        assert [
+            mock.call(["systemctl", "stop", "ubuntu-advantage.service"])
+        ] == m_subp.call_args_list
+
+    @mock.patch(M_PATH + "LOG.warning")
+    def test_stop_warning(self, m_log_warning, m_subp):
+        err = util.ProcessExecutionError("cmd")
+        m_subp.side_effect = err
+        stop()
+        assert [
+            mock.call(["systemctl", "stop", "ubuntu-advantage.service"])
+        ] == m_subp.call_args_list
+        assert [mock.call(err)] == m_log_warning.call_args_list
```

Comment:
I think we can just use `pytest.mark.parametrize` here instead, since the test are really similar